src/applications/vaos/reducers/newAppointment.js

```diff
@@ -351,12 +354,17 @@ export default function formReducer(state = initialState, action) {
         loadingEligibility: false,
       };
     }
-    case START_DIRECT_SCHEDULE_FLOW: {
+    case START_DIRECT_SCHEDULE_FLOW:
       return {
         ...state,
         pastAppointments: action.appointments,
+        flowType: FLOW_TYPES.DIRECT,
```

Comment:
Added `flowType` to state, so that `reasonForAppointment` knows which calendar page to go back to on `previous`.